### DISCUSSION TITLE
OBS Studio to KVS

### DISCUSSION THREAD

**Discussion by rainish2000** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1032] [Q&A]
I have a live camera feed that is currently received on OBS Studio. I want to then stream this to KVS. I have tried setting my stream output to rtsp://localhost:8554/live using the OBS Studio plugin `rtsp-server` and then ran the `kvs_gstreamer_sample` in the cpp library with the same localhost rtsp url, but kvs is not receiving anything and the Media Playback window in the KVS Console is blank. I know the credentials are correct because a new stream does get created if one does not exist with the name i give when running the sample. Is a local RTSP server even a possible way to stream to KVS?

Could there also be any alternatives that might be easier than RTSP? I've only tried RTSP because it had the most documentation on using it with KVS.

Id greatly appreciate any assistance in getting this OBS -> Kinesis flow to work, because it is a much bigger hassle accessing the camera feed directly because its heavily isolated and a LOT of overhead would be required if OBS cannot be used at all and the system in place has to be changed. 

edit: here is the program output when i run `kvs_gstreamer_sample`

[INFO ] [13-07-2023 06:45:42:553.709 GMT] Using region: ap-southeast-1
[INFO ] [13-07-2023 06:45:42:553.762 GMT] Using aws credentials for Kinesis Video Streams
[INFO ] [13-07-2023 06:45:42:553.811 GMT] No session token was detected.
[INFO ] [13-07-2023 06:45:42:554.593 GMT] createKinesisVideoClient(): Creating Kinesis Video Client
[INFO ] [13-07-2023 06:45:42:554.646 GMT] heapInitialize(): Initializing native heap with limit size 134217728, spill ratio 0% and flags 0x00000001
[INFO ] [13-07-2023 06:45:42:554.672 GMT] heapInitialize(): Creating AIV heap.
[INFO ] [13-07-2023 06:45:42:554.727 GMT] heapInitialize(): Heap is initialized OK
[DEBUG] [13-07-2023 06:45:42:554.869 GMT] getSecurityTokenHandler invoked
[DEBUG] [13-07-2023 06:45:42:554.911 GMT] Refreshing credentials. Force refreshing: 0 Now time is: 1689230742554900062 Expiration: 0
[INFO ] [13-07-2023 06:45:42:554.938 GMT] New credentials expiration is 1689234342
[INFO ] [13-07-2023 06:45:42:554.992 GMT] createDeviceResultEvent(): Create device result event.
[DEBUG] [13-07-2023 06:45:42:555.022 GMT] clientReadyHandler invoked
[DEBUG] [13-07-2023 06:45:42:555.098 GMT] Client is ready
[INFO ] [13-07-2023 06:45:42:555.192 GMT] Creating Kinesis Video Stream test1
[INFO ] [13-07-2023 06:45:42:555.230 GMT] createKinesisVideoStream(): Creating Kinesis Video Stream.
[INFO ] [13-07-2023 06:45:42:555.239 GMT] logStreamInfo(): SDK version: e8125ee363c4e77e11b15398a8b77f4e8a5791a5
[DEBUG] [13-07-2023 06:45:42:555.279 GMT] logStreamInfo(): Kinesis Video Stream Info
[DEBUG] [13-07-2023 06:45:42:555.325 GMT] logStreamInfo():      Stream name: test1
[DEBUG] [13-07-2023 06:45:42:555.374 GMT] logStreamInfo():      Streaming type: STREAMING_TYPE_REALTIME
[DEBUG] [13-07-2023 06:45:42:555.421 GMT] logStreamInfo():      Content type: video/h264
[DEBUG] [13-07-2023 06:45:42:555.452 GMT] logStreamInfo():      Max latency (100ns): 600000000
[DEBUG] [13-07-2023 06:45:42:555.481 GMT] logStreamInfo():      Fragment duration (100ns): 20000000
[DEBUG] [13-07-2023 06:45:42:555.542 GMT] logStreamInfo():      Key frame fragmentation: Yes
[DEBUG] [13-07-2023 06:45:42:555.567 GMT] logStreamInfo():      Use frame timecode: Yes
[DEBUG] [13-07-2023 06:45:42:555.591 GMT] logStreamInfo():      Absolute frame timecode: Yes
[DEBUG] [13-07-2023 06:45:42:555.615 GMT] logStreamInfo():      Nal adaptation flags: 0
[DEBUG] [13-07-2023 06:45:42:555.645 GMT] logStreamInfo():      Average bandwith (bps): 4194304
[DEBUG] [13-07-2023 06:45:42:555.676 GMT] logStreamInfo():      Framerate: 25
[DEBUG] [13-07-2023 06:45:42:555.707 GMT] logStreamInfo():      Buffer duration (100ns): 1200000000
[DEBUG] [13-07-2023 06:45:42:555.737 GMT] logStreamInfo():      Replay duration (100ns): 400000000
[DEBUG] [13-07-2023 06:45:42:555.766 GMT] logStreamInfo():      Connection Staleness duration (100ns): 600000000
[DEBUG] [13-07-2023 06:45:42:555.796 GMT] logStreamInfo():      Store Pressure Policy: 1
[DEBUG] [13-07-2023 06:45:42:555.821 GMT] logStreamInfo():      View Overflow Policy: 1
[DEBUG] [13-07-2023 06:45:42:555.849 GMT] logStreamInfo():      Segment UUID: NULL
[DEBUG] [13-07-2023 06:45:42:555.875 GMT] logStreamInfo():      Frame ordering mode: 0
[DEBUG] [13-07-2023 06:45:42:555.899 GMT] logStreamInfo(): Track list
[DEBUG] [13-07-2023 06:45:42:555.923 GMT] logStreamInfo():      Track id: 1
[DEBUG] [13-07-2023 06:45:42:555.947 GMT] logStreamInfo():      Track name: kinesis_video
[DEBUG] [13-07-2023 06:45:42:555.971 GMT] logStreamInfo():      Codec id: V_MPEG4/ISO/AVC
[DEBUG] [13-07-2023 06:45:42:555.995 GMT] logStreamInfo():      Track type: TRACK_INFO_TYPE_VIDEO
[DEBUG] [13-07-2023 06:45:42:556.019 GMT] logStreamInfo():      Track cpd: NULL
[INFO ] [13-07-2023 06:45:42:636.662 GMT] writeHeaderCallback(): RequestId: 1d254a63-6674-4746-9d04-e5cb9efb7f64
[DEBUG] [13-07-2023 06:45:42:636.738 GMT] describeStreamCurlHandler(): DescribeStream API response: {"StreamInfo":{"CreationTime":1.689177775984E9,"DataRetentionInHours":2,"DeviceName":"Kinesis_Video_Device","IngestionConfiguration":null,"KmsKeyId":"arn:aws:kms:ap-southeast-1:319877608589:alias/aws/kinesisvideo","MediaType":"video/h264","Status":"ACTIVE","StreamARN":"arn:aws:kinesisvideo:ap-southeast-1:319877608589:stream/test1/1689177775984","StreamName":"test1","Version":"kJNQMrneAOgqXl1Ahsjf"}}
[INFO ] [13-07-2023 06:45:42:637.380 GMT] describeStreamResultEvent(): Describe stream result event.
[INFO ] [13-07-2023 06:45:42:715.911 GMT] writeHeaderCallback(): RequestId: 3f2cba91-e736-4d8f-b514-b0abb523f8d2
[DEBUG] [13-07-2023 06:45:42:715.980 GMT] getStreamingEndpointCurlHandler(): GetStreamingEndpoint API response: {"DataEndpoint":"https://s-670061c1.kinesisvideo.ap-southeast-1.amazonaws.com"}
[INFO ] [13-07-2023 06:45:42:716.469 GMT] getStreamingEndpointResultEvent(): Get streaming endpoint result event.
[DEBUG] [13-07-2023 06:45:42:716.506 GMT] getStreamingTokenHandler invoked
[DEBUG] [13-07-2023 06:45:42:716.537 GMT] Refreshing credentials. Force refreshing: 1 Now time is: 1689230742716534763 Expiration: 1689234342
[INFO ] [13-07-2023 06:45:42:716.563 GMT] New credentials expiration is 1689234342
[INFO ] [13-07-2023 06:45:42:716.578 GMT] getStreamingTokenResultEvent(): Get streaming token result event.
[DEBUG] [13-07-2023 06:45:42:716.626 GMT] streamReadyHandler invoked
[DEBUG] [13-07-2023 06:45:42:716.802 GMT] Stream is ready
[INFO ] [13-07-2023 06:45:42:722.737 GMT] Streaming from rtsp source
New pad found: recv_rtp_src_0_583865637_96
New pad found: recv_rtp_src_1_2935074841_97
[INFO ] [13-07-2023 06:45:49:777.645 GMT] Pad link failed
[INFO ] [13-07-2023 06:45:49:777.652 GMT] Pad link successful

**Comment by sirknightj** [comment]
It should work with a local rtsp server. I am able to upload media using https://github.com/bluenviron/mediamtx

Could you please provide the full command you're using to run the sample?

**Reply by sirknightj** [reply]
Also, if you could also confirm that the rtsp url has media coming out of it? You could do that using vlc.

**Reply by sirknightj** [reply]
Also note: `rtspsrc` may have better results if you use `0.0.0.0` or `127.0.0.1` instead of `localhost`.
